stackrox/stackrox
# ROX-11918 Operator / Helm chart must allow to change the Route hostname

## Description

To allow a custom hostname for the `central` route.

## Checklist

PR in openshift-docs: https://github.com/openshift/openshift-docs/pull/48792

If any of these don't apply, please comment below.

## Testing Performed

To test helm updates:
```bash
cd stackrox/templates/helm/stackrox-central
helm install -n stackrox stackrox-central-services . --set-string central.exposure.route.enabled=true --set-string central.exposure.route.host=custom-host.apps.openshift.example.com
```
And observe that the `central` Route is created with the hostname specified.

To test OLM updates:
1. Create a bundle image with the updated CRD (`quay.io/akrohg/acs-bundle`)
2. Create a catalog source to house the bundle image (`quay.io/akrohg/acs-catalog`)
3. Create a `CatalogSource` in the `openshift-marketplace` namespace which points to the catalog image from step 2
4. Create a Central object and observe the new field, set some value for `spec.central.exposure.route.host`
5. Ensure that `central` route has a `host` attribute which matches the value set in step 4
<img width="824" alt="image" src="https://user-images.githubusercontent.com/5431427/182978397-079a4635-408a-42b0-96e3-08d6d97a8308.png">


In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.
